### PR TITLE
Rewind the request buffer after parsing the form.

### PR DESCRIPTION
**Pull request by Tratcher**
#366 
@muratg @BrennanConroy

### CODE REVIEW

**Comment by muratg** [comment]
:shipit:

**Comment by BrennanConroy** [comment]
:shipit:

**Comment by davidfowl** [comment]
Why Travis fail

File: src/Microsoft.AspNet.Http/Features/FormFeature.cs [hunk 1]
```diff
@@ -152,6 +152,9 @@ public IFormCollection ReadForm()
                 }
             }
 
+            // Rewind so later readers don't have to.
+            _request.Body.Seek(0, SeekOrigin.Begin);
```
Comment (kichalla) [inline src/Microsoft.AspNet.Http/Features/FormFeature.cs]: curious should we check if the request body is seekable?...like one could disable request buffering with `IHttpBufferingFeature`

Comment (Tratcher) [inline src/Microsoft.AspNet.Http/Features/FormFeature.cs]: The request body was made seekable on line 89 above.

**Comment by Tratcher** [comment]
@davidfowl Restore hung.

File: src/Microsoft.AspNet.Http/Features/FormFeature.cs (hunk 1 above)
Comment (kichalla) [inline src/Microsoft.AspNet.Http/Features/FormFeature.cs]: ah ok, thanks!

**Pull request merged**